### PR TITLE
Fix pfctl being invoked when NAT is not used + change ip var to ip4 

### PR DESCRIPTION
**Pull request by tschettervictor**
This PR fixes an issue where pfctl is invoked even when NAT is not being used. This prints unecessary error for folks using only VNET jails.

It also changes command to find ip4.addr to the bastille native config command which will return "not set" if ip4.addr is not found, instead of returning a -

### CODE REVIEW

**Comment by tschettervictor** [comment]
#346

**Comment by tschettervictor** [comment]
I also changed the _ip to _ip4 as it only greps for the ip4.addr value.
This will allow us to implement similar functions for future ip6 usage.

**Review by michael-o — CHANGES_REQUESTED** [review]
start.sh suffers from the same problem.

**Comment by tschettervictor** [comment]
Give it a go now.

File: usr/local/share/bastille/stop.sh [hunk 1]
```diff
@@ -52,10 +52,10 @@ for _jail in ${JAILS}; do
     ## test if running
     if [ "$(/usr/sbin/jls name | awk "/^${_jail}$/")" ]; then
         ## Capture ip4.addr address while still running
-        _ip="$(/usr/sbin/jls -j ${_jail} ip4.addr)"
+        _ip4="$( bastille config ${_jail} get ip4.addr )"
 
         # Check if pfctl is present
-        if which -s pfctl; then
+        if [ which -s pfctl ] && [ "${_ip4}" != "not set" ]; then
```
Comment (michael-o) [inline usr/local/share/bastille/stop.sh:58]: If start does not test for `pfctl`, stop should neither I guess. Otherwise `which` use is inconsistent...

Comment (tschettervictor) [reply usr/local/share/bastille/stop.sh]: Good point

File: usr/local/share/bastille/stop.sh [hunk 2]
```diff
@@ -52,10 +52,10 @@ for _jail in ${JAILS}; do
     ## test if running
     if [ "$(/usr/sbin/jls name | awk "/^${_jail}$/")" ]; then
         ## Capture ip4.addr address while still running
-        _ip="$(/usr/sbin/jls -j ${_jail} ip4.addr)"
+        _ip4="$( bastille config ${_jail} get ip4.addr )"
```
Comment (michael-o) [inline usr/local/share/bastille/stop.sh:55]: Why the space padding when other instance don't pad?

**Review by michael-o — APPROVED** [review]
WFM:
```
root@deblndw013x:/usr/local/share/bastille
# patch -p5 </tmp/bastille-patch
Hmm...  Looks like a unified diff to me...
The text leading up to this was:
--------------------------
|From aa17f5c4f91a1bc3779d86fe616c92f7823e858b Mon Sep 17 00:00:00 2001
|From: tschettervictor <85497460+tschettervictor@users.noreply.github.com>
|Date: Tue, 10 Dec 2024 14:51:29 -0700
|Subject: [PATCH 1/4] Fix pfctl being invoked when NAT is not used + change ip
| var to ip4 for future ip6 implementation
|
|---
| usr/local/share/bastille/stop.sh | 8 ++++----
| 1 file changed, 4 insertions(+), 4 deletions(-)
|
|diff --git a/usr/local/share/bastille/stop.sh b/usr/local/share/bastille/stop.sh
|index a3a8dfbe..ade6f9a6 100644
|--- a/usr/local/share/bastille/stop.sh
|+++ b/usr/local/share/bastille/stop.sh
--------------------------
Patching file stop.sh using Plan A...
Hunk #1 succeeded at 52.
Hunk #2 succeeded at 73.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|
|From 42a5a38334229def87a0e02018607a57d4052765 Mon Sep 17 00:00:00 2001
|From: tschettervictor <85497460+tschettervictor@users.noreply.github.com>
|Date: Tue, 10 Dec 2024 14:59:02 -0700
|Subject: [PATCH 2/4] fix start.sh also
|
|---
| usr/local/share/bastille/start.sh | 10 +++++-----
| 1 file changed, 5 insertions(+), 5 deletions(-)
|
|diff --git a/usr/local/share/bastille/start.sh b/usr/local/share/bastille/start.sh
|index f9e5a180..2eeb9e49 100644
|--- a/usr/local/share/bastille/start.sh
|+++ b/usr/local/share/bastille/start.sh
--------------------------
Patching file start.sh using Plan A...
Hunk #1 failed at 79.
1 out of 1 hunks failed--saving rejects to start.sh.rej
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|
|From 2a8a0702516be30f90024bbe0d21aad456a50639 Mon Sep 17 00:00:00 2001
|From: tschettervictor <85497460+tschettervictor@users.noreply.github.com>
|Date: Tue, 10 Dec 2024 15:04:27 -0700
|Subject: [PATCH 3/4] remove "which pfctl"
|
|We assume that if the jail has an ip4.addr value, then pfctl is obviously installed.
|
|It is also not invoked by start, so stop should not need it either.
|---
| usr/local/share/bastille/stop.sh | 2 +-
| 1 file changed, 1 insertion(+), 1 deletion(-)
|
|diff --git a/usr/local/share/bastille/stop.sh b/usr/local/share/bastille/stop.sh
|index ade6f9a6..d61d7ed6 100644
|--- a/usr/local/share/bastille/stop.sh
|+++ b/usr/local/share/bastille/stop.sh
--------------------------
Patching file stop.sh using Plan A...
Hunk #1 succeeded at 55.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|
|From 108227f72ee977d2cd7df1096caf72db23718675 Mon Sep 17 00:00:00 2001
|From: tschettervictor <85497460+tschettervictor@users.noreply.github.com>
|Date: Tue, 10 Dec 2024 15:07:29 -0700
|Subject: [PATCH 4/4] remove padding
|
|---
| usr/local/share/bastille/stop.sh | 2 +-
| 1 file changed, 1 insertion(+), 1 deletion(-)
|
|diff --git a/usr/local/share/bastille/stop.sh b/usr/local/share/bastille/stop.sh
|index d61d7ed6..6c4b7c1d 100644
|--- a/usr/local/share/bastille/stop.sh
|+++ b/usr/local/share/bastille/stop.sh
--------------------------
Patching file stop.sh using Plan A...
Hunk #1 succeeded at 52.
done
root@deblndw013x:/usr/local/share/bastille
# bastille restart deblndw013x2j
[deblndw013x2j]:
deblndw013x2j: removed

[deblndw013x2j]:
e0a_bastille3
e0b_bastille3
deblndw013x2j: created
```

**Comment by michael-o** [comment]
@bmac2 Can you merge this one as well?

**Comment by bmac2** [comment]
tested and works as described.  No longer get pfctl error on vnet jail on restart.

**Comment by michael-o** [comment]
> tested and works as described. No longer get pfctl error on vnet jail on restart.

Perfect,  now only a release for next quarterly is necessary.

**Pull request merged**